catalystwan/tests/test_feature_profile_api.py

```diff
@@ -38,73 +44,94 @@ class TestSystemFeatureProfileAPI(unittest.TestCase):
-
+    def test_get_method_with_valid_arguments(self, parcel, expected_path):
         # Act
-        api.get_parcels(self.profile_uuid, parcel, self.parcel_uuid)
+        self.api.get_parcels(self.profile_uuid, parcel, self.parcel_uuid)
 
         # Assert
-        mock_endpoint.get_by_id.assert_called_once_with(self.profile_uuid, expected_path, self.parcel_uuid)
+        self.mock_endpoint.get_by_id.assert_called_once_with(self.profile_uuid, expected_path, self.parcel_uuid)
 
     @parameterized.expand(endpoint_mapping.items())
-    @patch("catalystwan.session.ManagerSession")
-    @patch("catalystwan.endpoints.configuration.feature_profile.sdwan.system.SystemFeatureProfile")
-    def test_get_all_method_with_valid_arguments(self, parcel, expected_path, mock_endpoint, mock_session):
-        # Arrange
-        api = SystemFeatureProfileAPI(mock_session)
-        api.endpoint = mock_endpoint
+    def test_get_all_method_with_valid_arguments(self, parcel, expected_path):
+        # Act
+        self.api.get_parcels(self.profile_uuid, parcel)
 
+        # Assert
+        self.mock_endpoint.get_all.assert_called_once_with(self.profile_uuid, expected_path)
+
+    @parameterized.expand(endpoint_mapping.items())
+    def test_create_method_with_valid_arguments(self, parcel, expected_path):
         # Act
-        api.get_parcels(self.profile_uuid, parcel)
+        self.api.create_parcel(self.profile_uuid, parcel)
 
         # Assert
-        mock_endpoint.get_all.assert_called_once_with(self.profile_uuid, expected_path)
+        self.mock_endpoint.create.assert_called_once_with(self.profile_uuid, expected_path, parcel)
 
     @parameterized.expand(endpoint_mapping.items())
-    @patch("catalystwan.session.ManagerSession")
-    @patch("catalystwan.endpoints.configuration.feature_profile.sdwan.system.SystemFeatureProfile")
-    def test_create_method_with_valid_arguments(self, parcel, expected_path, mock_endpoint, mock_session):
-        # Arrange
-        api = SystemFeatureProfileAPI(mock_session)
-        api.endpoint = mock_endpoint
+    def test_update_method_with_valid_arguments(self, parcel, expected_path):
+        # Act
+        self.api.update_parcel(self.profile_uuid, parcel, self.parcel_uuid)
 
+        # Assert
+        self.mock_endpoint.update.assert_called_once_with(self.profile_uuid, expected_path, self.parcel_uuid, parcel)
+
+
+top_level_service_parcels = [
+    (
+        "dhcp-server",
+        LanVpnDhcpServerParcel(
+            parcel_name="DhcpServerDefault",
+            parcel_description="Dhcp Server Parcel",
+            address_pool=AddressPool(
+                network_address=Global[IPv4Address](value=IPv4Address("10.0.0.2")),
+                subnet_mask=Global[SubnetMask](value="255.255.255.255"),
+            ),
+        ),
+    )
+]
+
+
+class TestServiceFeatureProfileAPI(unittest.TestCase):
+    def setUp(self):
+        self.profile_uuid = UUID("054d1b82-9fa7-43c6-98fb-4355da0d77ff")
```

Comment:
can we generate random id with uuid4() function for test purposes?
This string might be detected as secret by pre-commit or github scanner as it has high entropy

Reply:
Yes, I'll change that